substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1978,6 +1906,32 @@ pub mod pallet {
 			}
 			Ok(Some(T::WeightInfo::deprecate_controller_batch(controllers.len() as u32)).into())
 		}
+
+		/// Forcefully Unbonds by Chilling first
+		/// Emits `Unbonded`.
+		#[pallet::call_index(29)]
+		#[pallet::weight(
+		T::WeightInfo::withdraw_unbonded_kill(SPECULATIVE_NUM_SPANS).saturating_add(T::WeightInfo::unbond()))
+		]
+		pub fn force_unbond(
+			origin: OriginFor<T>,
+			#[pallet::compact] value: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let controller = ensure_signed(origin)?;
+
+			let ledger = Self::ledger(StakingAccount::Controller(controller.clone()))?;
+
+			Self::chill_stash(&ledger.stash);
```

Comment:
It should only chill if unbond will fail without chilling right? i.e. when unbonding would make the bonded balance go below `MinNominatorBond`.

Also, please add tests.

Reply:
I have added the check for full unbonding.

i have also included test